statline_bq/utils.py

```diff
@@ -100,17 +107,69 @@ def get_metadata_cbs(urls: dict, odata_version: str) -> dict:
     ValueError
         If odata_version is not "v3" or "v4"
     """
-
     if odata_version == "v3":
-        target_url = "?".join((urls["TableInfos"], "$format=json"))
-        meta = requests.get(target_url).json()["value"][0]
+        catalog_urls = {
+            True: "https://dataderden.cbs.nl/ODataCatalog/Tables?$format=json",
+            False: "https://opendata.cbs.nl/ODataCatalog/Tables?$format=json",
+        }
+        url = catalog_urls[third_party]
+        params = {}
+        params["$filter"] = f"Identifier eq '{id}'"
+        tables = requests.get(url, params).json()["value"]
+        if len(tables) == 1:
+            meta = tables[0]
+        elif len(tables) == 0:
+            # TODO: HANDLE PROPERLY - is ValueError appropriate here?
```

Comment:
@galamit86 

Pythonic way to check whether a list is not empty is to use the fact that an empty list is falsy:

```
if tables: # truthy when list not empty
    do something
else:
    pass
````

Don't use `ValueErrror`

Reply:
Thanks! pythonizing the code, we have the following:
```
        if tables:
            if len(tables) == 1:
                meta = tables[0]
            else:
                pass
                # TODO
                # This means more then 1 result came back for the same ID - which is unlikely, and suggests a bug in the code more than anything.
        else:
            pass
            # TODO
            # This means no results came back - a possible scenario that is likely due to misspelling, or the dataset does not exist.
```
But I still need to do something in both else sections - getting more than 1 result, and getting 0 result. If not `ValueError` - any thoughts on an appropriate Exception, or different handling?

Reply:
I can't think of a scenario that len(tables) >=  1, so perhaps we could just leave it?

Reply:
Will do. I'm leaving the if/else section there with a `pass` though, just in case we ever want to debug something there.